components/main_page/commons/main_page_layout.lua

```diff
@@ -0,0 +1,58 @@
+---
```

Comment:
would it possibly be a good idea to support passing a module & functionName as part of content
for cases where we can just direct call stuff instead of having to preprocess

also maybe be able to pass widgets in the content?

Reply:
Sounds like a good future step